src/Queue/ServicesTrait.php

```diff
@@ -4,23 +4,32 @@
 namespace Queue\Queue;
 
 use Cake\Core\ContainerInterface;
+use LogicException;
 
 trait ServicesTrait {
 
 	/**
-	 * @var \Cake\Core\ContainerInterface
+	 * @var \Cake\Core\ContainerInterface|null
 	 */
-	protected ContainerInterface $container;
+	protected ?ContainerInterface $container = null;
 
 	/**
-	 * @param string $id Classname or identifier of the service you want to retrieve
+	 * @template OBJECT
+	 *
+	 * @param class-string<OBJECT> $id Classname or identifier of the service you want to retrieve
 	 *
-	 * @throws \Psr\Container\NotFoundExceptionInterface
 	 * @throws \Psr\Container\ContainerExceptionInterface
+	 * @throws \Psr\Container\NotFoundExceptionInterface
 	 *
-	 * @return mixed
+	 * @return OBJECT
```

Comment:
Why the uppercase?

Reply:
This is just a "random" name for the generic PHPStan template, not the php type `object`. It Could also be called `INSTANCE` or `T` for template. 

https://phpstan.org/blog/generics-in-php-using-phpdocs#type-variables

Reply:
It just means that an instance of the class XYZ is returned if you pass XYZ's class-string to it as an argument.